assets/src/hooks/use-end-date-change-listener.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External imports
+ */
+import { useCallback } from '@wordpress/element';
+import { getServerDateTime } from '@eventespresso/utils';
+import { ServerDateTime } from '@eventespresso/value-objects';
+import { isModelEntity } from '@eventespresso/validators';
+
+/**
+ * Verifies that end date value has changed and updates entity accordingly.
+ *
+ * @function
+ * @param {Object} props
+ * @member {Object} entity
+ * @member {Object} dateProps
+ * @member {string} startDateFormKey identifier for React Final Form data schema
+ * @member {string} endDateFormKey identifier for React Final Form data schema
+ * @member {Function} updateField callback for editing a field
+ * @member {Function} touchField callback for marking field as changed
+ * @return {Object} entityStartDate & entityEndDate
+ */
+const useEndDateChangeListener = ( {
+	entity,
+	dateProps,
+	startDateFormKey,
+	endDateFormKey,
+	updateField,
+	touchField,
+} ) => {
+	if ( ! isModelEntity( entity ) ) {
+		throw new TypeError(
+			'Invalid Entity supplied to useStartDateChangeListener'
+		);
+	}
+	return useCallback( ( newDateValue, prevDateValue ) => {
+		if ( newDateValue && newDateValue !== prevDateValue ) {
+			const newDate = getServerDateTime( newDateValue );
+			if ( newDate instanceof ServerDateTime ) {
+				entity[ dateProps.end ] = newDate;
+			}
+			touchField( startDateFormKey );
+			touchField( endDateFormKey );
+		}
+	}, [
+		entity[ dateProps.start ],
+		entity[ dateProps.end ],
+		startDateFormKey,
+		endDateFormKey,
+		updateField,
+		touchField,
```

Comment:
Why should it depend upon `updateField` and `touchField`?

Reply:
from the React documentation regarding dependencies:

> The array of dependencies is not passed as arguments to the function. Conceptually, though, that’s what they represent: every value referenced inside the function should also appear in the dependencies array.

My understanding of that is that unless you are 1000% certain that some external code used by a hook is completely static and will not change, then it should be included as a dependency (just to be clear I was not using the term "static" here to mean "a class function not belonging to an instance", but just to mean something that stays the same).

I don't know if `updateField` and `touchField` are completely unchanging and therefore I have included them as dependencies. It's probably fair to assume that those callbacks won't change as they are likely just simple setters... but I'm just not 1000% certain of that...

Reply:
👍🏻